### ISSUE TITLE
Add support for fine-grained permissions personal token

### ISSUE THREAD

**Issue by luki-** [nathonius/obsidian-github-link#79]
Hey, I'd love to use this plugin to see what is the status of a Pull Request - is it still open, approved, or merged?
Currently, looks like the plugin requires the repository contents permissions, but maybe it would be possible to give it only a pull request read, to check the metadata without having access to the code?

**Comment by nathonius** [comment]
@luki- this is a good idea. I'm curious, have you already tried with a token without that scope? It may just work at the moment, I don't think I'm actually hitting any APIs which use that scope yet.

I can definitely build features going forward such that scope are optional, or maybe configurable?

**Comment by luki-** [comment]
@nathonius yes, I tried it yesterday and unfortunately it didn't work. I added the Issues (read-only),Pull requests (read-only), along with a required metadata permission and the plugin didn't replace my private repos URLs with the badge, but continued to work for public repositories. I couldn't find any logs in the devtools, I'm not sure if there's something else I should enable to view them there.

**Comment by nathonius** [comment]
With the latest version of the plugin, you can enable debug logging in plugin settings. Every request and response to github's api will be logged to the console. Perhaps there's an error there?

I'll test this on my end when I get to it, but if you could share any error logs from there that would be great.

**Comment by drsm79** [comment]
I'm not sure if this is the same, but I'm getting a "Request failed, status 422" message in the log when accessing a private repo with a fine grained access token. The token is configured as follows for the repo in question.
![Screenshot 2024-05-29 at 13 06 35](https://github.com/nathonius/obsidian-github-link/assets/104090/1ce881eb-1639-4ee9-819e-add48fc5dafc)

**Comment by drsm79** [comment]
Seems to be when the query is `queryType: pull-request`

**Comment by drsm79** [comment]
Hmm, no, PEBKAC - fixed with the right query.

**Comment by nathonius** [comment]
@luki- I cannot reproduce this issue. I created a fine-grained PAT with just the pull request read permission and was able to resolve a PR from a private repo. So I'm guessing you were just missing the org setting. For example, for my private repos, I added `nathonius` in the Orgs and users setting:
![image](https://github.com/nathonius/obsidian-github-link/assets/4851889/8b70fc4d-55e0-45c5-abcd-ab51d01e9c98)
